src/services/MineSweeperService.ts

```diff
@@ -0,0 +1,77 @@
+import NumberUtils from "../utils/NumberUtils";
+
+class MineSweeperService {
+	private static instance: MineSweeperService;
+
+	/* eslint-disable */
+	private constructor() { }
+	/* eslint-enable */
+
+	static getInstance(): MineSweeperService {
+		if (!this.instance) {
+			this.instance = new MineSweeperService();
+		}
+
+		return this.instance;
+	}
+
+	public generateGame(rows: number, cols: number, ratio: number): string {
+		const bombCount = rows * cols / ratio;
+		const bombIndex = [];
+		const numbers = [":zero:", ":one:", ":two:", ":three:", ":four:", ":five:", ":six:", ":seven:", ":eight:"];
+		const positions = [[0, -1], [0, 1], [-1, -1], [-1, 0], [-1, 1], [1, -1], [1, 0], [1, 1]];
+		const bomb = ":boom:";
+
+		const grid = Array.from(Array(rows), () => new Array(cols).fill(0));
+
+		function isOutsideEdges(rowIndex: number, colIndex: number) {
```

Comment:
Don't declare a function within a method - make this a private method on the class.